test/sockets.test.js

```diff
@@ -1,63 +1,186 @@
 import 'regenerator-runtime/runtime';
 
 import Socket from '../src/sockets';
-import Logger from '../src/logger';
+import { Logger, SOCKET_PING } from 'syft-helpers.js';
 
 import { WebSocket, Server } from 'mock-socket';
 
 global.WebSocket = WebSocket;
 
 const url = 'ws://localhost:8080/';
 
-const onMessage = event => console.log('NEW MESSAGE', event);
-const onOpen = event => console.log('NEW OPEN', event);
-const onClose = event => console.log('NEW CLOSE', event);
-const onError = event => console.log('NEW ERROR', event);
+/**
+ * Creates promise from the first event call
+ * @param emitter Object that emits the event
+ * @param event Event name
+ * @returns {Promise}
+ */
+function makeEventPromise(emitter, event) {
```

Comment:
Another thing: what does this function actually do?

Reply:
This is for convenience. It wraps event callback in a promise. The promise is resolved when the event is triggered.
E.g. when we want to await for established connection on the mock server:
`const socket = await makeEventPromise(mockServer, 'connection');`